From c16d6d46dfdeff509b888140d4fa1a04888ca355 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 20 Jan 2012 09:06:08 -0500
Subject: [PATCH] XSettings: be more careful when shutting down

The shell name watch_id may still be 0 when we shut down right
away, because another instance is already running.
---
 plugins/xsettings/gsd-xsettings-manager.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/plugins/xsettings/gsd-xsettings-manager.c b/plugins/xsettings/gsd-xsettings-manager.c
index 075cb37..fa49a3e 100644
--- a/plugins/xsettings/gsd-xsettings-manager.c
+++ b/plugins/xsettings/gsd-xsettings-manager.c
@@ -929,7 +929,8 @@ gnome_xsettings_manager_stop (GnomeXSettingsManager *manager)
 
         stop_fontconfig_monitor (manager);
 
-        g_bus_unwatch_name (manager->priv->shell_name_watch_id);
+        if (manager->priv->shell_name_watch_id > 0)
+                g_bus_unwatch_name (manager->priv->shell_name_watch_id);
 
         if (p->settings != NULL) {
                 g_hash_table_destroy (p->settings);
-- 
1.7.8.4

